File: python/bad_compression/bad_compression_test.py
# coding=utf-8
from bad_compression import bad_compression


def test_empty():
    assert bad_compression("") == ""


def test_a():
    assert bad_compression("a") == "a"


def test_aba():
    assert bad_compression("aba") == "aba"


def test_aa():
    assert bad_compression("aa") == ""


def test_aab():
    assert bad_compression("aab") == "b"


def test_aaaaa():
    assert bad_compression("aaaaa") == "a"


def test_aabb():
    assert bad_compression("aabb") == ""


def test_abba():
    assert bad_compression("abba") == ""


def test_abaaba():
    assert bad_compression("abaaba") == ""


def test_aaabccddd():
    assert bad_compression("aaabccddd") == "abd"

File: java/bad_compression/src/test/java/com/devartis/exercise/bad_compression/BadCompressionTest.java
package com.devartis.exercise.bad_compression;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class BadCompressionTest {

    @Test
    public void test_empty() {
        assertEquals("", Utils.badCompress(""));
    }

    @Test
    public void test_a() {
        assertEquals("a", Utils.badCompress("a"));
    }

    @Test
    public void test_aba() {
        assertEquals("aba", Utils.badCompress("aba"));
    }

    @Test
    public void test_aa() {
        assertEquals("", Utils.badCompress("aa"));
    }

    @Test
    public void test_aab() {
        assertEquals("b", Utils.badCompress("aab"));
    }

    @Test
    public void test_aaaaa() {
        assertEquals("a", Utils.badCompress("aaaaa"));
    }

    @Test
    public void test_aabb() {
        assertEquals("", Utils.badCompress("aabb"));
    }

    @Test
    public void test_abba() {
        assertEquals("", Utils.badCompress("abba"));
    }

    @Test
    public void test_abaaba() {
        assertEquals("", Utils.badCompress("abaaba"));
    }

    @Test
    public void test_aaabccddd() {
        assertEquals("abd", Utils.badCompress("aaabccddd"));
    }
}